src/m365/spo/commands/site/site-remove.ts

```diff
@@ -95,240 +109,149 @@ class SpoSiteRemoveCommand extends SpoCommand {
-            }
+        if (siteDetails.TimeDeleted === null) {
+          throw `Site ${args.options.url} is not in the recycle bin. Remove it first.`;
+        }
 
-            await this.deleteGroup(group.id, logger);
-            await this.deleteSite(args.options.url, args.options.wait, logger);
+        if (isGroupSite) {
+          const isGroupPermanentlyDeleted = await this.isGroupPermanentlyDeleted(siteDetails.GroupId, logger);
+          if (!isGroupPermanentlyDeleted) {
+            await this.removeGroupFromEntraRecycleBin(siteDetails.GroupId, logger);
           }
-          catch (err: any) {
-            if (this.verbose) {
-              await logger.logToStderr(`Site group doesn't exist. Searching in the Microsoft 365 deleted groups.`);
-            }
-
-            const deletedGroups = await this.isSiteGroupDeleted(groupId);
-            if (deletedGroups.value.length === 0) {
-              if (this.verbose) {
-                await logger.logToStderr("Site group doesn't exist anymore. Deleting the site.");
-              }
-
-              if (args.options.wait) {
-                await logger.logToStderr(chalk.yellow(`Entered site is a groupified site. Hence, the parameter 'wait' will not be applicable.`));
-              }
-
-              await this.deleteOrphanedSite(logger, args.options.url);
-            }
-            else {
-              throw `Site group still exists in the deleted groups. The site won't be removed.`;
-            }
+        }
+        await this.deleteSiteFromSharePointRecycleBin(args.options.url, logger);
+      }
+      else {
+        if (siteDetails.TimeDeleted !== null) {
+          throw `Site ${args.options.url} is already in the recycle bin. Use --fromRecycleBin to permanently delete it.`;
+        }
+
+        if (isGroupSite) {
+          await this.deleteGroupifiedSite(args.options.url, logger);
+          if (args.options.skipRecycleBin) {
+            await this.removeGroupFromEntraRecycleBin(siteDetails.GroupId, logger);
           }
         }
+        else {
+          await this.deleteNonGroupSite(args.options.url, logger);
+        }
+
+        if (args.options.skipRecycleBin) {
+          await this.deleteSiteFromSharePointRecycleBin(args.options.url, logger);
+        }
       }
     }
     catch (err: any) {
       this.handleRejectedPromise(err);
     }
   }
 
-  private isSiteGroupDeleted(groupId: string): Promise<{ value: { id: string }[] }> {
-    const requestOptions: any = {
-      url: `https://graph.microsoft.com/v1.0/directory/deletedItems/Microsoft.Graph.Group?$select=id&$filter=groupTypes/any(c:c+eq+'Unified') and startswith(id, '${groupId}')`,
-      headers: {
-        accept: 'application/json;odata.metadata=none'
-      },
-      responseType: 'json'
-    };
-
-    return request.get<{ value: { id: string }[] }>(requestOptions);
-  }
+  private async removeGroupFromEntraRecycleBin(groupId: string, logger: Logger): Promise<void> {
```

Comment:
For this function, and all others, let's use `logger` as first argument to keep it consistent.